cloudnetwork/v1/types.go

```diff
@@ -0,0 +1,75 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CloudPrivateIPConfig performs assignments of a private IP address to the
+// primary NIC associated with cloud VMs. This is done by specifying the IP and
+// Kubernetes node which the IP should be assigned to. This CRD is intended to
+// be used by the network plugin which manages the cluster network. The spec
+// side represents the desired state requested by the network plugin, and the
+// status side represents the current state that this CRD's controller has
+// executed. No users will have permission to modify it, and if a cluster-admin
+// decides to edit it for some reason, their changes will be overwritten the
+// next time the network plugin reconciles the object. Note: the CR's name
+// must specify the requested private IP address (can be IPv4 or IPv6).
+// The IP format will be validated in an admission controller.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=cloudprivateipconfig,scope=Cluster
+type CloudPrivateIPConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// spec is the definition of the desired private IP request.
+	// +kubebuilder:validation:Required
+	// +required
+	Spec CloudPrivateIPConfigSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	// status is the observed status of the desired private IP request. Read-only.
+	// +kubebuilder:validation:Optional
+	// +optional
+	Status CloudPrivateIPConfigStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// CloudPrivateIPConfigStatus specifies the requested private IP, its corresponding node assignment,
+// together with its assignment condition.
+// +k8s:openapi-gen=true
+type CloudPrivateIPConfigStatus struct {
+	// condition is the assignment condition of the private IP and its status
+	// +kubebuilder:validation:Required
+	// +required
+	Condition metav1.Condition `json:"condition" protobuf:"bytes,2,opt,name=condition"`
```

Comment:
`Conditions []metav1.Condition` plural if you would

Reply:
I am not sure what multiple conditions we could have, given this new atomic approach? What we mentioned in the previous version of this API (here: https://github.com/openshift/api/pull/859#discussion_r600886940), I think can be covered by one sole status now:

```
apiVersion: cloud.network.openshift.io/v1
kind: CloudPrivateIPConfig
metadata:
  name: 192.168.126.11
spec:
  node: nodeX
status:
  conditions:
  - message: ""
    reason: ""
    status: "True|False|Unkown"
    type: Assigned
```

According to what I mentioned here: https://github.com/openshift/api/blob/40ed3a6c48dab348e227785d8d499e38110fa275/cloudnetwork/v1/types.go#L58-L62

Reply:
If you start with one, you cannot compatibly grow to more than one.  Also, the kube standard across nearly every resource is plural, so things like `oc wait` only work with the slice.